src/ifp/src/InitFloorplan.cc

```diff
@@ -416,7 +426,22 @@ void InitFloorplan::makeUniformRows(odb::dbSite* base_site,
 
   auto make_rows = [&](dbSite* site) {
     const uint site_dy = site->getHeight();
-    const int rows_y = core_dy / site_dy;
+    int rows_y = core_dy / site_dy;
+
+    switch (row_parity) {
+      case RowParity::NONE:
+        break;
+      case RowParity::EVEN:
+        rows_y = (rows_y / 2) * 2;
+        break;
+      case RowParity::ODD:
+        if (rows_y > 0) {
+          rows_y = ((rows_y - 1) / 2) * 2 + 1;
```

Comment:
Fwiw this works but I find `rows_y = rows_y % 2 ? rows_y - 1 : rows_y` more obvious

Reply:
Thanks, I agree, I've changed it.